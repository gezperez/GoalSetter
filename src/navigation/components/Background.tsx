import React, { FC, memo } from 'react'
import { Dimensions, ImageBackground, ImageStyle, StyleSheet } from 'react-native'

import { Images } from '~/utils'

const WIDTH = Dimensions.get('window').width

type BackgroundProps = {
  height: number
}

const Background: FC<BackgroundProps> = ({ height }) => (
  <ImageBackground source={Images.HEADER_BACKGROUND} style={[styles.background, { height }]} />
)

type Styles = {
  background: ImageStyle
}

const styles = StyleSheet.create<Styles>({
  background: {
    width: WIDTH,
  },
})

export default memo(Background)
