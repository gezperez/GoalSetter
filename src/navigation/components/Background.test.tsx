import React from 'react'
import { render } from '@testing-library/react-native'

import Background from './Background'

describe('Background', () => {
  afterEach(jest.resetAllMocks)

  it('should render with required props', () => {
    const { toJSON } = render(<Background height={100} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
