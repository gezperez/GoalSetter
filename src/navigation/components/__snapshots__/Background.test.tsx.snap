// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background should render with required props 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "width": 750,
      },
      Object {
        "height": 100,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/background.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
</View>
`;
